h5py/h5t.pyx

```diff
@@ -1642,7 +1637,7 @@ cpdef TypeID py_create(object dtype_in, bint logical=0, bint aligned=0):
             return _c_complex(dt)
 
         # Compound
-        elif (kind == c'V') and (dtype_in.names is not None):
+        elif (kind == c'V') and (getattr(dt, "names") is not None):
```

Comment:
Is `getattr()` necessary? `dt` should always be a dtype object, no?

Reply:
`dt` gives access to the C struct of dtype, and in there, `names` is either a tuple or a NULL pointer instead of `None` (see https://github.com/numpy/numpy/blob/master/numpy/core/include/numpy/ndarraytypes.h#L639).
The use of `getattr` makes sure `names` it is accessed through the Python API in order to get `None`.
An alternative way of writing it is casting dt to a python object first: `((<object> dt).names != None)`
And an equivalent check but written the C-like way is  `(<void *> dt.names != NULL)`.
(Those alternatives work).